modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/AbstractCollocationBenchmark.java

```diff
@@ -0,0 +1,141 @@
+
+    /** {@code true} enables collocation feature. */
+    private static final Boolean ENABLE_COLLOCATION_FEATURE = true;
+
+    protected final List<KeyValueView<Tuple, Tuple>> tableViews = new ArrayList<>();
+
+    @Override
+    protected int nodes() {
+        return 1;
+    }
+
+    @Override
+    protected int replicaCount() {
+        return 1;
+    }
+
+    @Override
+    protected String clusterConfiguration() {
+        if (tinySchemaSyncWaits()) {
+            return super.clusterConfiguration();
+        } else {
+            // Return a magic string that explicitly requests production defaults.
+            return PRODUCTION_CLUSTER_CONFIG_STRING;
+        }
+    }
+
+    @Override
+    protected void createDistributionZoneOnStartup() {
+        ZoneDefinition zone = ZoneDefinition.builder(SHARED_ZONE_NAME)
+                .partitions(partitionCount())
+                .replicas(replicaCount())
+                .storageProfiles(DEFAULT_STORAGE_PROFILE)
+                .build();
+
+        publicIgnite.catalog().createZone(zone);
+    }
+
+    @Override
+    protected void createTableOnStartup() {
+        for (int i = 1; i <= tableCount(); ++i) {
+            TableDefinition tableDefinition = TableDefinition.builder("test_table_" + i)
+                    .columns(
+                            column("id", ColumnType.INTEGER),
+                            column("company", ColumnType.varchar(32)))
+                    .primaryKey("id")
+                    .zone(SHARED_ZONE_NAME)
+                    .build();
+
+            Table t = publicIgnite.catalog().createTable(tableDefinition);
+
+            tableViews.add(t.keyValueView());
+        }
+    }
+
+    @Override
+    public void nodeSetUp() throws Exception {
+        // Enable/disable collocation feature.
+        System.setProperty(FEATURE_FLAG_NAME, ENABLE_COLLOCATION_FEATURE.toString());
+
+        // Start the cluster and initialize it.
+        super.nodeSetUp();
+
+        // Patch replica manager to propagate table replication messages to zone replication groups.
+        if (ENABLE_COLLOCATION_FEATURE) {
+            int catalogVersion = igniteImpl
+                    .catalogManager()
+                    .latestCatalogVersion();
+
+            int zoneId = igniteImpl
+                    .catalogManager()
+                    .catalog(catalogVersion)
+                    .zone(SHARED_ZONE_NAME.toUpperCase())
+                    .id();
+
+            igniteImpl.replicaManager().groupIdConverter(request -> {
+                if (!(request instanceof WriteIntentSwitchReplicaRequest)) {
+                    if (request.groupId().asReplicationGroupId() instanceof TablePartitionId) {
+                        TablePartitionId tablePartitionId = (TablePartitionId) request.groupId().asReplicationGroupId();
+
+                        return new ZonePartitionId(zoneId, tablePartitionId.partitionId());
```

Comment:
Should tableId also be passed to the constructor?

Reply:
No, it should not. I asked @kgusakov about it, and he confirmed that the `tableId` should be ignored, at least for now.